External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -267,184 +267,31 @@
-    
-                % In this subfunction, we determine the difference within the current column
-                % Which we can use for extrapolation, and to check if we need to exclude edge voxels
-                % of the current column
-                [diffIs, minInd, maxInd, thresholdIs] = xASL_im_Extrapolate_FixEdges(imThisCol, 1);
-
-                % Now we check if we should remove edge voxels
-                % We try this nEdgeVoxels times (default=5)
-                nLayersProcessed = 0;
-                while nLayersProcessed<nEdgeVoxels
-                    % 
-                    if diffIs(end)>thresholdIs
-                        imThisCol(maxInd) = NaN; % remove the edge voxel
-                        % If we have changed a first or last non-nan voxel, we rerun the difference determination 
-                        [diffIs, minInd, maxInd, thresholdIs] = xASL_im_Extrapolate_FixEdges(imThisCol, 1);
-                    end
-                    nLayersProcessed = nLayersProcessed+1;
-                end
-
-                meanDiff = xASL_stat_MeanNan(diffIs);
-
-                % Create a robust last value, from 6 voxels
-                % This avoids extrapolating an extreme value at the edge
-                voxels6 = imThisCol(maxInd-5:maxInd); % select the last 6 voxels
-                voxels6 = voxels6(isfinite(voxels6)); % select those without NaNs
-                robustMedian = median(voxels6);
-                robustDiff = median(voxels6(2:end) - voxels6(1:end-1)); % mean difference
-                numVoxels = length(voxels6)/2;
-                robustValue = robustMedian + numVoxels*robustDiff;
-                % Now take the mean of the actual last value and the robust one
-                robustValue = mean([robustValue imThisCol(maxInd)]);
-
-                % Apply this
-                imThisCol(maxInd+1:end) = robustValue + (1:nCol-maxInd) * meanDiff;
-                % Put this column back in the image
-                imCol(:, ColumnIs) = imThisCol;
-            end
-    
-            %% 4. Reshape 2D back to 3D
-            switch(DimensionIs)
-                case 1
-                    IM = reshape(imCol, [nX,nY,nZ]);
-                case 2
-				    % For 2D matrix and 3D matrix, the reshaping back is done differently
-				    if ndims(IM)<3
-					    IM = shiftdim(reshape(imCol,[nY,nX]),1);
-				    else
-					    IM = shiftdim(reshape(imCol,[nY,nZ,nX]),2);
-				    end
-                case 3
-                    IM = reshape(imCol', [nX,nY,nZ]);
-            end % switch
-        end % for DimensionIs
-    end % sum(isnan(IM))
-end
-
-% Any remaining NaNs had to be inside, so we interpolate them (setting them
-% to zeros creates artifacts!)
-IM = xASL_im_ExtrapolateSmoothOverNaNs(IM, 0);
-
-end
+%% Extrapolation of values based on the edge values 
+% Calculate the distance map from the non-Nan towards NaN
+[~, distY, distX, distZ] = xASL_im_DistanceTransform(~maskNaN);
```

Comment:
very weird that this is not X Y Z but Y X Z :)

Reply:
Fixed. See above.

Reply:
See above.